x-pack/plugins/cases/server/services/user_actions/transform.ts

```diff
@@ -0,0 +1,316 @@
+  [UserActionFieldType.New]: CONNECTOR_ID_REFERENCE_NAME,
+  [UserActionFieldType.Old]: USER_ACTION_OLD_ID_REF_NAME,
+};
+
+interface ExtractedConnector {
+  transformedActionDetails: unknown;
+  references: SavedObjectReference[];
+}
+
+function transformCreateConnector(
+  connector: CaseConnector,
+  fieldType: UserActionFieldType
+): ExtractedConnector {
+  const { transformedConnector, references } = transformConnectorIdToReference(
+    ConnectorIdReferenceName[fieldType],
+    connector
+  );
+
+  return {
+    transformedActionDetails: transformedConnector,
+    references,
+  };
+}
+
+type ConnectorIdRefNameType =
+  | typeof CONNECTOR_ID_REFERENCE_NAME
+  | typeof USER_ACTION_OLD_ID_REF_NAME;
+
+export const transformConnectorIdToReference = (
+  referenceName: ConnectorIdRefNameType,
+  connector?: {
+    id?: string;
+  }
+): {
+  transformedConnector: { connector: unknown };
+  references: SavedObjectReference[];
+} => {
+  const { id: connectorId, ...restConnector } = connector ?? {};
+
+  const references = createConnectorReference(connectorId, ACTION_SAVED_OBJECT_TYPE, referenceName);
+
+  const { id: ignoreNoneId, ...restNoneConnector } = getNoneCaseConnector();
+  const connectorFieldsToReturn =
+    connector && references.length > 0 ? restConnector : restNoneConnector;
+
+  return {
+    transformedConnector: {
+      connector: connectorFieldsToReturn,
+    },
+    references,
+  };
+};
+
+const createConnectorReference = (
+  id: string | null | undefined,
+  type: string,
+  name: string
+): SavedObjectReference[] => {
+  return id && id !== noneConnectorId
+    ? [
+        {
+          id,
+          type,
+          name,
+        },
+      ]
+    : [];
+};
+
+function isUpdateCaseConnector(
+  action: string,
+  actionFields: string[],
+  actionDetails: unknown
+): actionDetails is CaseConnector {
+  try {
+    return isUpdateConnector(action, actionFields) && CaseConnectorRt.is(actionDetails);
+  } catch {
+    return false;
+  }
+}
+
+function transformUpdateConnector(
+  connector: CaseConnector,
+  fieldType: UserActionFieldType
+): ExtractedConnector {
+  const { transformedConnector, references } = transformConnectorIdToReference(
+    ConnectorIdReferenceName[fieldType],
+    connector
+  );
+
+  return {
+    transformedActionDetails: transformedConnector.connector,
+    references,
+  };
+}
+
+type CaseExternalService = rt.TypeOf<typeof CaseExternalServiceBasicRt>;
+
+function isPushConnector(
+  action: string,
+  actionFields: string[],
+  actionDetails: unknown
+): actionDetails is CaseExternalService {
+  try {
+    return isPush(action, actionFields) && CaseExternalServiceBasicRt.is(actionDetails);
+  } catch {
+    return false;
+  }
+}
+
+export const PushConnectorIdReferenceName: Record<
+  UserActionFieldType,
+  PushConnectorIdRefNameType
+> = {
+  [UserActionFieldType.New]: PUSH_CONNECTOR_ID_REFERENCE_NAME,
```

Comment:
These references are specific for a push user action. We probably could get away with using the normal ones defined above but I just wanted to keep with the same reference name pattern that the cases and configuration saved objects use.